compute/ibm_vpc.py

```diff
@@ -0,0 +1,451 @@
+
+        node = None
+        while end_time > datetime.now():
+            sleep(5)
+            node = self.service.get_instance(instance_id)
+            node_details = node.get_result()
+
+            if node_details["status"] == "running":
+                end_time = datetime.now()
+                duration = (end_time - start_time).total_seconds()
+                LOG.info(
+                    "%s moved to running state in %d seconds.",
+                    node_details["name"],
+                    int(duration),
+                )
+                return
+
+            if node_details["status"] == "error":
+                msg = (
+                    "Unknown Error"
+                    if not node.extra
+                    else node.extra.get("fault").get("message")
+                )
+                raise NodeError(msg)
+
+        raise NodeError(f"{node_details['name']} is in {node_details['status']} state.")
+
+    def wait_until_node_del(self, instance_id):
+        """
+        Wait till the VM deleted
+        Args:
+            instance_id    Id of node instance
+        """
+        try:
+            start_time = datetime.now()
+            end_time = start_time + timedelta(seconds=1200)
+
+            node = None
+            while end_time > datetime.now():
+                sleep(5)
+                node = self.service.get_instance(instance_id)
+                node_details = node.get_result()
+                if not node:
+                    end_time = datetime.now()
+                    duration = (end_time - start_time).total_seconds()
+                    LOG.info(
+                        "%s moved to running state in %d seconds.",
+                        node_details["name"],
+                        int(duration),
+                    )
+                    return
+            raise NodeError(f"{node.name} is in {node.state} state.")
+        except ApiException as ae:
+            if ae.code == 404:
+                return
+            raise NodeError("Unable to delete the node")
+
+    def clean_up_instances_ibm(self, node_name, timeout):
+        """
+        Method to clean up instances in ibm cloud.
+        Args:
+            node_name    Name of node instance
+            timeout      Max time to wait for the deletion to complete
+        """
+        pattern = node_name
+        instance_list = self.service.list_instances()
+        del_instance_list = [
+            i for i in instance_list.get_result()["instances"] if pattern in i["name"]
+        ]
+        with parallel() as p:
+            for instance in del_instance_list:
+                LOG.info(
+                    "Destroying node {node_name} with {timeout} timeout".format(
+                        node_name=instance["name"], timeout=timeout
+                    )
+                )
+                p.spawn(self.service.delete_instance, instance["id"])
+                self.wait_until_node_del(instance["id"])
```

Comment:
IMO, this method definition would be in`utils.utils` . Secondly, we should wait for the node to be delete either inside `delete_instance` or outside of the `for loop`... when doing outside of the `for loop`, it would need to check for all the nodes in the list